src/com/owncloud/android/ui/activity/ShareActivity.java

```diff
@@ -116,25 +116,42 @@ protected void onNewIntent(Intent intent) {
         }
     }
 
-    private void doShareWith(String shareeName, boolean isGroup) {
+    private void doShareWith(String shareeName, String dataAuthority) {
+
+        int lastPointIndex = dataAuthority.lastIndexOf(".");
+        String type = dataAuthority.substring(lastPointIndex + 1);
+        ShareType shareType = null;
+        switch (type) {
```

Comment:
I thought it twice. Your proposal, @AndyScherzinger, is indeed better.

The reason of using ```dataAuthority``` is that the contract of search providers does not allow return arbitrary parameters in a search. Otherwise we would return directly ```shareType```. We are simply "tricking" the search provider interface.

The responsibility to "translate the trick" would be better in ```UserAndGroupsSearchProvider``` than in its clients (in this case, ```ShareActivity```) . So, would be better to create the map you suggest in ```UserAndGroupsSearchProvider```, and a static public method that allow ```ShareActivity``` get the ```ShareType``` corresponding to the received URL.